File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:

        left = 0
        right = len(heights) - 1
        max_water = 0

        while left < right:
            width = right - left
            length = 0

            if heights[left] < heights[right]:
                length = heights[left]
                left += 1
            else:
                length = heights[right]
                right -= 1
            max_water = max(max_water, width * length)
        
        return max_water

        

File: Data Structures & Algorithms/is-palindrome/submission-0.js
class Solution {
  /**
   * @param {string} s
   * @return {boolean}
   */
  isPalindrome(s) {
    const n = s.length;
    let left = 0;
    let right = n - 1;

    while (left < right) {
      if (
        !(
          s[left].toLowerCase().charCodeAt() >= 48 &&
          s[left].toLowerCase().charCodeAt() <= 57
        ) &&
        !(
          s[left].toLowerCase().charCodeAt() >= 97 &&
          s[left].toLowerCase().charCodeAt() <= 122
        )
      ) {
        left++;
      } else if (
        !(
          s[right].toLowerCase().charCodeAt() >= 48 &&
          s[right].toLowerCase().charCodeAt() <= 57
        ) &&
        !(
          s[right].toLowerCase().charCodeAt() >= 97 &&
          s[right].toLowerCase().charCodeAt() <= 122
        )
      ) {
        right--;
      } else {
        if (s[left].toLowerCase() !== s[right].toLowerCase()) {
          return false;
        } else {
          left++;
          right--;
        }
      }
    }

    return true;
  }
}
